### ISSUE TITLE
Support display options in embedded mode

### ISSUE THREAD

**Issue by zstadler** [IsraelHikingMap/Site#361]
Currently, there is no way to set the language of an embedded map.
Shared maps do not preserve the language in which the sharing link was created, and there is no way to otherwise set the default language.

**Comment by HarelM** [comment]
I think the whole concept of embedded maps needs to be revised.
Currently you can't configure which buttons to show and which not to show, which features to support and add you mentioned the language.
I believe that current embedding is bad and I think it will be better to remove it as an option until we decide how to properly solve it and what are our requirements in terms of what other sites (our clients) need and how...

**Comment by zstadler** [comment]
https://en.wikipedia.org/wiki/Perfect_is_the_enemy_of_good

Embedding is already in use by several sites. Removing it or changing it in a non-backward-compatible way isn't good customer/partner orientation.

**Comment by HarelM** [comment]
True, although I don't know who are those clients to inform them on any changes we do.
I can't really stop supporting embedded mode and it's not what I meant when I said we should remove it - what I meant is remove the instruction from the share dialog (the iframe instructions are pretty simple to anyone who invested a few hours on building a  website).
Also allowing more flexibility and definitions in embedded mode doesn't have to break anything in theory and we can think how to implement it so that it won't break in the future (as much as possible).
Much like `s=<shar id>` or `url=<file-address>`, we can add `embedded=<embedded-options>` and describe how to use it in a wiki page here.
What do you think?
I'll send a mail to those that I know are embedding us and ask them to join this discussion.

I agree that perfect is the enemy of good, but in this case, this feature requirement is just partial to what our clients need, I think.

**Comment by zstadler** [comment]
For reference, and for future testing, here is an example of embedding:
[Yarok Balev](http://www.yarokbalev.org/%d7%98%d7%91%d7%a2-%d7%91%d7%97%d7%99%d7%a4%d7%94/%d7%9e%d7%a4%d7%95%d7%aa/%d7%9e%d7%a4%d7%aa-%d7%94%d7%98%d7%99%d7%95%d7%9c%d7%99%d7%9d/).
Note: this is and old embedding, which is currently broken.

**Comment by HarelM** [comment]
This is related to the change with the "!" it's not related to embedded mode directly.
Probably need to be supported again, maybe part of issue #352.

**Comment by HarelM** [comment]
But as I said, We need to know who our customer are, if we do not know we can't inform them on changes we make. 
I agree we should reduce the number of changes that are not backward compatible as much as possible but we need to be able to do breaking changes when needed...

**Comment by YarokBalev** [comment]
We're a very satisfied client of your project :-)

**Comment by HarelM** [comment]
That's great news! But it looks like the link in your site as mentioned by @zstadler is not presenting what you want, which was caused by us.
While this can be easily fixed I'm interested to know what would you like to present and not present in your site when using IHM in embedded mode.
For example, there are no zoom-in and zoom-out buttons in the link above, this is because the screen in smaller and the site thinks it's presenting itself on mobile where you can have pitch-zoom using gestures.
So basically, I want to know what are the relevant options for you and what are the non relevant.

**Comment by MatanBenMoshe** [comment]
I used the tails instead of iframe because the zoom and fitBounds features are missing.
It cause the trail and interest points to be shown in unfriendly way- part of the trail was out of the screen and not centered.
Not sure how many customers are interested in this feature. The tails solution is good for me.

**Comment by HarelM** [comment]
I'm guessing that the zoom is missing because the iframe is small and we removed the zoom buttons on small screens (assuming they have touch-pitch) but we haven't thought about embedded mode so this is more of a bug than anything else.
@MatanBenMoshe - About the center of a trail, can you send me a link so I can check why, this sounds like another bug (please report them so we can fix them, we don't have any QA so we relay on users to let us know when something is not working :-)).
In any case, as this feature suggests, since you are no longer using an iframe, what would be the minimal features requirements in order to change your site back to using the iframe.
I image that you'd need the following capabilities: zoom, route statistics, layers pane to be able to see google earth, save to file in order to allow the user to save the route to file, print(?).

**Comment by MatanBenMoshe** [comment]
Using iframe has advantages and disadvantages.
The main disadvantage on my side is that I have partially control on the data and the design shown on my web. It means: 
  -  If I want to change something in design (For example the route color) I need to go over all the links and change it.
  -  Bugs and problems can appear on my page without any change on my website.
  -  I can't use the rich Google Maps API in the map design. (Foe example the circle moving on the route)

The big advantage is all the features you have on your website. All the features you mentioned are relevant. print can be also very nice.
I'll try to make web page with iframe that will present the center issue.

**Comment by HarelM** [comment]
I totally agree about your perspective (although the same goes for third-party packages that you use in your site, except you have just a bit more control whether to upgrade a version or not).
In theory there's a third option: creating leaflet plug-ins from the code in this repository.
This should facilitate everyone's needs since you could add the relevant plug-ins to your site while still have full control of the code.
Unfortunately, there's no good UI framework in leaflet which makes the work on plug-ins that have a UI component a total nightmare.
This also requires a lot of help coding from people who would like to use the code I wrote in their site and I must admit I have very little motivation to do it (not that I'll mind helping but still).
To summarize, I can relate to the approach of not using an iframe.
If you decide to consider using an iframe please use this thread to define an API to use with the site according to your needs.

**Comment by YarokBalev** [comment]
It seems that the iframe on our website works well now. Thanks for the update :-)
http://www.yarokbalev.org/טבע-בחיפה/מפות/מפת-הטיולים/

Is there a way to embed this map in a chosen language, or should we write a cumbersome explanation for the English speakers (or any other language that the data would be translated to)?

**Comment by HarelM** [comment]
Right now you can't do it.
The main purpose of this thread is to understand what are the embedded mode requirements and implement them.
So far all I got is:
 * Zoom buttons should be visible.
 * Add the ability to support landuage.
I can implement the above two, but I prefer to implement something more robust. 
In order to implement something more robust I need to see a variety of requirements and think of a design that will solve most or all of them.
Therefore, I need more requirements.

**Comment by valleyofdawn** [comment]
Another request, from BikePanel.com is for the height profile to be optionally open upon embedding.

**Comment by HarelM** [comment]
To summarize this thread a bit:
There are two types of clients: website developers and site owners who are not developers.
Since website developers need full control of the content of their web page this will not be addressed in this issue.
The requests that I see in this thread are the following:
1. Language selection.
2. Showing the height profile.
3. Controlling which buttons to present.

I suggest the following solution:
1. A dedicated screen/page to allow iframe users to create an iframe with the right content - this screen will show an iframe to allow the user to see what s/he selected and a form to allow selection of the elements to present.
2. Use the following contract in the address-bar to signal the site how it should look:
`/?version=v1&controls=[all|zoom|location|layers|files|save-as|info|edit-osm|profile|search|drawing|share|language|stats]&language=[he|en]&showstats=[true|false]`
3. nice-to-have : support `~` to signal what not to show to allow shorter address.

Things to consider:
 * Changing the language in the site stores this in local storage so we need to think if we want to the iframe to change local storage or not.
 * Presenting or not POI layers is not supported by the requirements above and currently uses the local storage.
 * IHM icon will be present and there won't be an option to remove it.
 * If the layers button is not visible it might affect the ability to show the layers - this needs to be tested.

**Comment by HarelM** [comment]
This has been open for more than two years. I don't see any value in implementing it.
If someone who is using the iframe would like to support our site by placing a bounty on this issue I might consider re-opening it.
